rust/common/src/value.rs

```diff
@@ -157,17 +157,62 @@ impl_tvm_context!(
     DLDeviceType_kDLExtDev: [ext_dev]
 );
 
+/// A struct holding TVM byte-array.
+///
+/// ## Example
+///
+/// ```
+/// let v = b"hello";
+/// let barr = TVMByteArray::from(&v);
+/// assert_eq!(barr.len(), v.len());
+/// assert_eq!(barr.data(), 7[104u8, 101, 108, 108, 111]);
+/// ```
 impl TVMByteArray {
     pub fn data(&self) -> &'static [u8] {
         unsafe { std::slice::from_raw_parts(self.data as *const u8, self.size) }
     }
+
+    /// Gets the length of the underlying byte-array
+    pub fn len(&self) -> usize {
+        self.size
+    }
+
+    /// Gets the underlying byte-array as `Vec<i8>`
+    pub fn to_vec(&self) -> Vec<i8> {
```

Comment:
`u8` is byte. `i8` is `char`. I know that the inner type is `i8`, but the name is wrong :P

Reply:
Yeah, that was a copy from frontend and I already had `data` there giving `Vec<i8>` but you defined it differently so I needed to change. I'll update the doc to `Converts the ... to Vec<i8>`.

Reply:
Can you please just make it `u8`? There's no case in which its more efficient or usable to make a _byte_ array a _char_ array.

Reply:
I believe I can not (which I annoying I know). Here's the def

```
pub struct TVMByteArray {
    pub data: *const ::std::os::raw::c_char,
    pub size: usize,
}
```
so `self.data` is `i8` so makes the whole thing into `Vec<i8>`. Unless, we can guarantee the `i8` to `u8` conversion is safe though!

Reply:
> guarantee the `i8` to `u8` conversion is safe though

yeah just `mem::transmute` it. There's no version of reality in which 8 bits  is not 8 bits :)